src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = {};
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        stateHistory.push({});
```

Comment:
Instead of pushing an empty object here, consider modifying the `currentState` and then pushing it outside of the `switch` block. This will ensure consistency with the rule that states a copy of the object should be pushed at the end of each loop cycle.